src/com/jsonnetplugin/JsonnetIdentifierReference.java

```diff
@@ -25,7 +25,25 @@ public JsonnetIdentifierReference(@NotNull PsiElement element, TextRange textRan
         String identifier = this.getElement().getText();
         PsiElement element = this.getElement();
         while (element != null) {
-            if (element instanceof JsonnetOuterlocal) {
+            if (element instanceof JsonnetSelect) {
+                List<JsonnetIdentifier0> selectList = new ArrayList<>();
+                for (JsonnetSelect select : ((JsonnetExpr)element.getParent()).getSelectList()) {
+                    selectList.add(select.getIdentifier0());
+                    if (select == element) {
+                        break;
+                    }
+                }
+                JsonnetObj obj = resolveExprToObj(
+                        (JsonnetExpr)element.getParent(),
+                        new ArrayList<>(),
+                        selectList
```

Comment:
When jumping to members, instead of breaking on the first JsonnetSelect node has INTELLIJ_RULES, we instead break on the first select node which matches the one the user is clicking on. I extracted the common logic into an overload of `resolveExprToObj`